### ISSUE TITLE
The abstraction around the UI configuration options is nonsensical

### ISSUE THREAD

**Issue by dio-gh** [RPCS3/rpcs3#10549]
![image](https://user-images.githubusercontent.com/10439488/125778494-11998d81-1a70-42dd-8970-d07f7fc0cf65.png)

Apparently, there are facilities to manage the current UI configuration and its lifecycle, which includes profile creation, saving, loading and resetting even.

.. Why is this a thing?

First off, nobody creates different UI profiles, since there's genuinely no meaningful reason to do so.

Secondly, I see zero compelling reasons as to why the configuration of the UI should be managed separately from the rest of the emulator, and by extension, why shouldn't there be a "reset to defaults" option for the emulator options as a whole.

It's the kind of feature that grips both too little and not enough at the same time.

The `UI Colors` are also a bit strange, why would that be managed separately from the rest of the colors in a given stylesheet? If it exists to make it easier for users to change these, why do they receive higher priority than literally every other color that makes up the entire emulator's UI? Pointless set of options.

Basically the entire tab could be obsoleted, by moving the `Log` groupbox to the `Advanced` tab, moving `UI Stylesheet`, `Discord Integration` and `UI Options` to the `Emulator` tab, and nuking the rest.

It's a bit like how the `Network` tab could also be completely merged into the `System` tab; or how the `Pad` options should really be just a regular tab within the rest of the settings.

**Comment by Megamouse** [comment]
It's the GUI tab because it does not modify the config.yml in any way and gets hidden during custom config dialogs.

The UI colors exist because users didn't like the way some icons looked in their preferred theme.
Good luck telling people to change their Stylesheets only to get it updated by the next auto update.

The config profiles exist because someone liked to quickly switch from their Multimonitor setup to some other setup. It remembers window states and so on.

**Comment by dio-gh** [comment]
> Good luck telling people to change their Stylesheets only to get it updated by the next auto update.

I've literally been rocking my own for a year now, that's the entire point of having stylesheets. Does require users to have enough braincells to copy the theme into a new file before adding their own modifications though.

> The config profiles exist because someone liked to quickly switch from their Multimonitor setup to some other setup. It remembers window states and so on.

I switch between multimonitor and single sometimes, and such a need never even came up to me. This is an OS issue to begin with, see the Windows 11 feature reveal about this exact thing. Was this some feature request? How does this actually make things simpler for the user? Seems like about equal of a drag to me on first read.

> and gets hidden during custom config dialogs.

This does make a lotta sense, good point, but the whole management system around it imo is still insanely overkill and misdirected, see above.

**Comment by Megamouse** [comment]
You can't expect every single user to write their own stylesheet whenever some new game has a bad icon. It's kinda ironic since you literally just opened a support issue asking for help with it.

Feel free to make a poll on the config profiles though.
I don't think I ever used them apart from debugging.
Anyway, this existed since the beginning of time and you're the first person who seems upset about it. 🦆

**Comment by dio-gh** [comment]
> It's kinda ironic since you literally just opened a support issue asking for help with it.

This is kind of in bad faith. I do have an ongoing issue with the styling capabilities of certain elements, but
 - it's not because I suck at this;
 - it's not because it's so difficult to do;
 - it's not even tangentially related to icon backgrounds.

Would you need further convincing, Goose and MsLow also have their own themes, and have had them since forever. By this logic, you could build an entire theming system that allows for a "more accessible theming of the UI for the end users". Like yeah nah.

It's also a bit weird how we've established that GUI settings are global-only, but now you mention "[one] new game has a bad icon", meaning you can easily end up in a situation where one or two games' icons just don't work, no matter what color you choose. Coincidentally, game icons can also be changed individually on their own without ever touching those settings, by the virtue of just flat out replacing them.

> and you're the first person who seems upset about it

I'm not that super crazed about it, it started from one of the everconfused-type users on Discord mentioning two days ago that they mistook the UI settings' `Reset to default` for an emulator-wide reset button. This kinda got me thinking why is this a thing to begin with, and so here we are.

Also, how do you imagine me setting up a poll? Where would I post it, how would I promote it? This kinda seems like a Patreon poll candidate, though I'd need to trust whoever posts the poll on my behalf to be impartial, and that the post would be open ended enough. I can ask staff I guess, if that's how you imagined it.

**Comment by Megamouse** [comment]
All of your points are valid. But you're missing the point with Stylesheets. They're supposed to be static as in you don't need to change them frequently unless you want to actually change the overall style.
The icon backgrounds can change every day depending on how many games you install. Users shouldn't ever have to touch Stylesheets just because one icon looks weird. That's why there are icon colors in order to make it simple to just change it on the fly.

Sure, now that icons can be changed we can consider removing the colors.

But again, this is just you and me debating on our preferences and opinions. Keep in mind that there might be a lot of people (or at least a very vocal minority) that actually like some of these features.
So just removing it might cause the issue to pop up again, just the other way around.

For a poll you might ask staff. Idk

**Comment by Megamouse** [comment]
There's one thing I forgot.
The GUI config reset is actually kinda useful, since there might be times where the game list and other lists get weird size issues due to strange Qt quirks, for example when updating your OS or changing some unknown OS parameters, or maybe even by a Qt update. This might manifest as missing or huge columns.

One way to solve this in extreme cases is to delete the ini file or to press the reset button.

**Comment by dio-gh** [comment]
I guess the icon color modification could actually be savored, if it was reorganized e.g. to here:

![image](https://user-images.githubusercontent.com/10439488/125908616-50dc214f-b0ce-4a9d-9e8b-04aa5e30558b.png)

It might even improve the discoverability of it. Would require renaming the main list item there from `Custom Images` to `Customize` though.

As for the reset, I'm not really sure that's not the greatest way to go about that. That kinda sucks though, I'm not sure how it could be resolved better. Issue is, that reset resets a bit too many things, so you can't even e.g. call it silently on every application launch and then reapply the stylesheet. Feels like a bit of a nuclear solution to me, never really had to use it myself either, or have seen it being advised for use. Guess it's more of a Linux thing? Sounds like to me these bugs should be individually checked and catered for, as much of a pain that may be.

I'll ask around staff if that's the final verdict then btw.

**Comment by Megamouse** [comment]
There seems to be some kind of confusion here.
The GUI window state and the Stylesheets are two entirely different things.

**Comment by dio-gh** [comment]
If i click reset, it tells me it will reset the stylesheet also. Is that incorrect then?

**Comment by Megamouse** [comment]
Well. I've prepared a branch that removes the config boxes.
So if you find a representative majority I can remove it whenever I want.

**Comment by dio-gh** [comment]
I'll ask around and attempt to get a Patreon poll organized, thanks.

**Comment by dio-gh** [comment]
> So if you find a representative majority I can remove it whenever I want.

Only managed to gather 20 responses through the patron-only poll due to internal politics sabotaging it, but here's the turnout:

![image](https://user-images.githubusercontent.com/10439488/127138306-58ff1ad8-bab1-4e30-926a-437025529c6c.png)

![image](https://user-images.githubusercontent.com/10439488/127138405-2e44501c-d7dc-464b-a902-dc1df0aebf9d.png)

I can extend the poll to Reddit if you want a wider audience to put their two cents in, but I figured this should be good enough.

There were also further questions in the poll, posting the results here in lieu of a better place:

<details><summary>long image, press here to open</summary>

![image](https://user-images.githubusercontent.com/10439488/127139766-8b07c200-5457-4a0c-b8fa-8c5ca15739bc.png)
</details>

Feel free to use them as inspiration for further changes. I'll do too.

**Comment by Megamouse** [comment]
I don't really think 20 people out of thousands or even more is representative

**Comment by dio-gh** [comment]
Alright, Reddit time then. I'd like to remind you though that if you're expecting thousands of fills, there's no audience or polling method I can reach to to achieve that. At most it'll be in the triple digits. The vblank poll plateaued at 135.

If you'd not be convinced by a hundred-some fill poll either, maybe we can add the problem to one of the great undecidability problems in computing, in the worst case :p

**Comment by dio-gh** [comment]
The Reddit questionnaire has concluded. Managed to gather an exceptionally high 200 fills flat, with the inclusion of Discord in the last couple days. While it is impossible to determine whether this turnout is genuinely statistically representative, both because of us not actually being statisticians, and also because of us not having the slightest clue how many people actually use RPCS3 due to the no data collection policy the project has on, this is the best data I that can be offered.

Same result.

![image](https://user-images.githubusercontent.com/10439488/128394246-e24ae845-4d11-4a6d-95b9-c77189a54c0d.png)

184 (92%) of people say that they've never used this feature. (The question was illustrated with a screenshot and marked with a red box, to ensure everyone was on the same page.)

![image](https://user-images.githubusercontent.com/10439488/128396774-c0d2ad21-cc81-429f-b0cb-94a60a027af9.png)

Curiously, 2 people from that group managed to mess up, and suggest that they *did* try the feature in the next question. In any case, the reasons given in the next question were the following:

![image](https://user-images.githubusercontent.com/10439488/128396513-44893dbf-c67d-4905-acde-cc254caf9028.png)

It's perfectly apparent that all of these people didn't manage to realize that the feature is separate from the stylesheet selection, meaning the feature not only was redundant and useless, but also caused confusion.

Moving on to the UI reset part of the problem:

![image](https://user-images.githubusercontent.com/10439488/128397338-635ec0ca-33be-43f1-b671-005fb3222204.png)

193 (96.5%) of people reported never having used the UI reset button, which makes sense, as this figure should be equal or higher than the previous. Of the remaining 7 people, 2 report that they've used it after being instructed to, and 5 report using it on their own.

![image](https://user-images.githubusercontent.com/10439488/128397980-c7914c7c-061e-4275-a1fa-d14e2b6d7a37.png)

It'd seem that 3 out of those 7 people were possibly filling the questionnaire at random. In the remaining 4 that didn't, there's also a person who's confused beyond saving, a person who considered giving a reason for their behavior too invasive, and a person who was having conspiracy theories about performance. These are also all effectively noise in the data.

Of the 2 actual issues remaining, one user was having a DPI scaling issue, and another encountered a weird setting persistence related bug. I don't think I can consider these blockers.

All in all, the feature seems safely nukable.
